components/inputs/demo/input-number.html

```diff
@@ -98,6 +98,18 @@ <h2>Trailing Zeroes</h2>
 				</template>
 			</d2l-demo-snippet>
 
+			<h2>Denominator</h2>
+			<d2l-demo-snippet>
+				<template>
+					<d2l-input-number max="5" min="0" label="Grade" label-hidden required unit="/5" unit-label="out of 5" value="3"></d2l-input-number>
```

Comment:
I was playing around with the value. Depending on the unit chosen, the consumer may very likely need to use `input-width`... since the default may not be sufficient.
![image](https://user-images.githubusercontent.com/9042472/206527989-6c5b399c-0cdb-4221-8837-c4df3302ef6d.png)

It looks like `d2l-input-percent` doesn't actually have enough width when the value is 100. 😞 . That's maybe separate from the issue at hand here.
![image](https://user-images.githubusercontent.com/9042472/206527704-f7a89a90-9bfe-48ce-ac85-d0ffc0171942.png)

Aside from that, I notice that the font-size for the value and the unit are different - 16px and 19px respectively. Looks a little odd.
![image](https://user-images.githubusercontent.com/9042472/206529050-67351189-c1d7-4362-85de-19dbfd9519e9.png)

Reply:
Changing the default width at this point may or may not be safe.

Reply:
I noticed the font-size thing too but assumed it was intentional. Could explain why `100` doesn't fit in the percent input. I'll investigate what happened there.

Reply:
Looking at the [original visual diff](https://github.com/BrightspaceUI/core/blob/1688dbcffdc34f91fad3dc6f56349e0ddcc02d94/components/inputs/test/screenshots/ci/golden/input-percent/d2l-input-percent-default-value.png), I think this was always an issue.

Having said that, I'm treating it as a bug since it's way more obvious now that the unit is multiple characters. I've updated the unit to also be `0.8rem`. Looks much better now:

<img width="371" alt="Screen Shot 2022-12-08 at 1 43 47 PM" src="https://user-images.githubusercontent.com/5491151/206544175-5a50e3ed-b301-435c-9e31-9ec4aed69e82.png">

Reply:
Going back to the width... would there be any risk to me making the `input-width` based on `ch` units. So if there are 2 digits in the denominator, we want to make room for at least 2 digits in the numerator. 
```
inputWidth = (scoreDenominator.toString().length * 2) + 4 // Add a little extra for padding and the "/"
...
input-width="${inputWidth}ch"
```
Ends up looking like this:
![image](https://user-images.githubusercontent.com/90270112/206549517-6fbf51f0-1a1e-4dbb-901d-8fa48314df2c.png)
![image](https://user-images.githubusercontent.com/90270112/206549560-8d8a7b2e-c490-4679-9512-f5b0bff086f2.png)
![image](https://user-images.githubusercontent.com/90270112/206549584-486d3fc1-1569-4ce4-8f98-3018a5574dc0.png)
![image](https://user-images.githubusercontent.com/90270112/206549612-64d2244d-5445-477c-98c5-bdfe3757019e.png)

Reply:
Hmm yeah. I think our default font's character width for numbers is identical, but I cant guarantee it'll be like that in all cases. I'll have to get creative here or just make it oversized and have it look less good.

Minor nit, but can we have the invalid icon float left when text is right aligned? It defaults to float right and can cut off text when there is a validation error.

Reply:
That's an excellent idea... but I think that'll require introducing an actual `alignment` attribute instead of relying on the CSS variable. :(

Reply:
I'll explore this separately if I have time tomorrow, but I'll merge this for now so you can start using it. It's tricky as-is because the validation icon already aligns itself in a weird way based on a combination of the unit width and the RTL-ness of the page.